lib/recurly/schema/json_parser.rb

```diff
@@ -61,16 +61,11 @@ def self.recurly_class(type)
         Resources::Page
       else
         type_camelized = type.split("_").map(&:capitalize).join
-        if Resources.const_defined?(type_camelized)
-          klazz = Resources.const_get(type_camelized)
-          if klazz.ancestors.include?(Resource)
-            klazz
-          else
-            if Recurly::STRICT_MODE
-              raise ArgumentError, "Could not find Recurly Resource responsible for key #{type}"
-            else
-              nil
-            end
+        if Resources.const_defined?(type_camelized, false)
+          Resources.const_get(type_camelized, false)
+        else
```

Comment:
What do you think about making this `elsif Recurly::STRICT_MODE` then it would not have an else clause (returning nil)?